sos/report/plugins/arcconf.py

```diff
@@ -21,11 +21,41 @@ class arcconf(Plugin, IndependentPlugin):
     commands = ("arcconf",)
 
     def setup(self):
-
-        # get list of adapters
-        self.add_cmd_output([
-            "arcconf getconfig 1",
-            "arcconf list",
-            "arcconf GETLOGS 1 UART"
-        ])
+        # Get the list of available adapters
+        listarcconf = self.collect_cmd_output("arcconf list")
+
+        # Parse the 'arcconf list' output and extract controller IDs.
+        # For each Controller ID found in 'arcconf list', add commands
+        # for getconfig and GETLOGS
+        #
+        # Sample 'arcconf list' output:
+        #
+        # Controller information
+        # -------------------------------------------------------------
+        # Controller ID   : Status, Slot, Mode, Name, SerialNumber, WWN
+        # -------------------------------------------------------------
+        # Controller 1:   : Optimal, Slot XXXX, XXXX, XXXX, XXXX, XXXX
+        # -------------------------------------------------------------
+        # Controller 2:   : Optimal, Slot XXXX, XXXX, XXXX, XXXX, XXXX
+
+        if listarcconf['status'] == 0:
+            for line in listarcconf['output'].splitlines():
+                words = line.split()
+
+                # Skip lines with fewer than two words
+                if len(words) < 2:
+                    continue
+
+                # Line with "Controller XX: *" has Controller ID
+                if words[0] != "Controller" or not words[1].endswith(":"):
+                    continue
```

Comment:
If we use a regex and matching groups here, we can simplify the line parsing and automatically get the `controller_id` as well.

E.G. (off the top of my head, may need some refining):

```python
try:
    controller_id = re.match(r"(^Controller (\d))", line).groups()[1]
    if controller_id:
        self.add_cmd_output([...])
except Exception:
    continue
```